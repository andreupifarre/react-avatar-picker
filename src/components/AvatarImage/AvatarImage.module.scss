.AvatarImage {
  position: relative;
  box-sizing: border-box;
  width: 60px;
  height: 60px;
  border-radius: 50%;
  display: flex;
  align-items: center;
  justify-content: center;
  cursor: pointer;
}

.AvatarImageImg {
  width: inherit;
  height: inherit;
  position: relative;
  overflow: hidden;
  border-radius: inherit;
  pointer-events: none;
}
