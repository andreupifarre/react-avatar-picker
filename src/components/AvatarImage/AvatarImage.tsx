import React from 'react';
import styles from './AvatarImage.module.scss';

type AvatarImageType = {
  id: number;
  src: string;
  label: string;
  size?: number;
  onClick: (id: number) => void;
};

const AvatarImage = ({
  id,
  src,
  label,
  size = 60,
  onClick,
}: AvatarImageType) => {
  const handleClick = () => {
    onClick(id);
  };

  return (
    <div
      className={styles.AvatarImage}
      style={{ width: `${size}px`, height: `${size}px` }}
      onClick={handleClick}
    >
      <img className={styles.AvatarImageImg} src={`/img/${src}`} alt={label} />
    </div>
  );
};

export default AvatarImage;
