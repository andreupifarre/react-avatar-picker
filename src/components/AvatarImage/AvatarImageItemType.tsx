export type AvatarImageItemType = {
  id: number;
  src: string;
  label: string;
};
