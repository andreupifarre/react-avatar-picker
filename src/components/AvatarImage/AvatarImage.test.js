import React from 'react';
import { shallow } from 'enzyme';

import AvatarImage from './AvatarImage';

const render = (props) => shallow(<AvatarImage {...props} />);

describe('AvatarImage', () => {
  it('should show the properties passed to the component, on the image', () => {
    const item = {
      src: 'avatar1.png',
      label: 'Avatar 1',
      id: 1,
      onClick: () => {},
    };

    const wrapper = render({ ...item });
    const image = wrapper.find('img');

    expect(image.prop('src')).toEqual('/img/' + item.src);
    expect(image.prop('alt')).toEqual(item.label);

    wrapper.simulate('click');
  });

  it('should render correctly', () => {
    expect(render().getElements()).toMatchSnapshot();
  });
});
