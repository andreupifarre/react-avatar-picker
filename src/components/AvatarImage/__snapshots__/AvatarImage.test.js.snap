// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvatarImage should render correctly 1`] = `
Array [
  <div
    className="AvatarImage"
    onClick={[Function]}
    style={
      Object {
        "height": "60px",
        "width": "60px",
      }
    }
  >
    <img
      className="AvatarImageImg"
      src="/img/undefined"
    />
  </div>,
]
`;
